### PR TITLE
UI Polish based on feedback

### PR DESCRIPTION
**Pull request by LibbyTang**
- [x] Issue: #20 
- [x] Issue: #21
- [x] Issue: #22 
- [x] Issue: #23
- [x] Issue: #26:  Apply the region filter to Amsterdam on both the routes and the stops.

Also make the map fly to the vehicle of a route when the route is selected.

### CODE REVIEW

**Comment by Bramdevriez** [comment]
Lots of nice features added here! 

Some feedback:
- The arrows indicating the route direction do not take into account route offset.
- Is there a need to be able to switch regions? Not sure about hard-coding in the north-holland ID.
- Sometimes it can take quite a while before the loading icon switches from 'Uninitiated". Is there a reason for this long wait before data is fetched?

Some stuff to think about for future improvements:
- I think the top-bar could use a bit more structure. Based on the icons alone it is not immediately clear whether they are buttons or indicators and what their functions are. The inconsistent spacing also makes it seem a bit disorganised. Some suggestions:
a. Move the "list of routes", routes filter and departure/destination search into the top of the routes list, so it is clear these belong together.
b. Group together the "general" items: connection icon, map settings, download, time.
c. Group total vehicles and vehicles filter.
- In some cases the selected route is hidden behind other routes, making it difficult to see what is selected. Would be nice to be able to view a single route with its stops and vehicles only.
Screenshot 2023-12-15 at 11 27 01

- Maybe the routes table could be condensed somewhat. Currently a lot of horizontal scrolling needs to happen to view all information. Maybe some of the information could be put on a secondary line.

Hope these notes don't overwhelm, just some thoughts. Happy to discuss more on discord.

File: src/components/Vehicles/VehiclePopupText.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+interface VehiclePopupTextProps {
+    vehicle: string;
+    route: PTRouteProperties;
+    delay: number;
+    timestamp: number;
+}
+
+const VehiclePopupText: React.FC<VehiclePopupTextProps> = ({ vehicle, route, delay, timestamp }) => {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const _ = require('lodash');
```
Comment (Bramdevriez) [inline src/components/Vehicles/VehiclePopupText.tsx:12]: Why is lodash required like this?

Comment (LibbyTang) [reply src/components/Vehicles/VehiclePopupText.tsx:12]: I just changed this by installing and importing this :)

**Comment by LibbyTang** [comment]
> Lots of nice features added here!
> 
> Some feedback:
> 
> * The arrows indicating the route direction do not take into account route offset.
> * Is there a need to be able to switch regions? Not sure about hard-coding in the north-holland ID.
> * Sometimes it can take quite a while before the loading icon switches from 'Uninitiated". Is there a reason for this long wait before data is fetched?
> 
> Some stuff to think about for future improvements:
> 
> * I think the top-bar could use a bit more structure. Based on the icons alone it is not immediately clear whether they are buttons or indicators and what their functions are. The inconsistent spacing also makes it seem a bit disorganised. Some suggestions:
>   a. Move the "list of routes", routes filter and departure/destination search into the top of the routes list, so it is clear these belong together.
>   b. Group together the "general" items: connection icon, map settings, download, time.
>   c. Group total vehicles and vehicles filter.
> * In some cases the selected route is hidden behind other routes, making it difficult to see what is selected. Would be nice to be able to view a single route with its stops and vehicles only.
> 
> Screenshot 2023-12-15 at 11 27 01
> * Maybe the routes table could be condensed somewhat. Currently a lot of horizontal scrolling needs to happen to view all information. Maybe some of the information could be put on a secondary line.
> 
> Hope these notes don't overwhelm, just some thoughts. Happy to discuss more on discord.

Thanks a lot for such detailed feedback!

**Review by Bramdevriez — APPROVED** [review]

**Pull request merged**